File: user/src/main/java/com/kjj/user/dto/user/UserDto.java
package com.kjj.user.dto.user;

import com.kjj.user.entity.User;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserDto {
    private Long id;
    private String username;
    private String password;
    private String roles;

    public static UserDto from(User user) {
        return new UserDto(
                user.getId(),
                user.getUsername(),
                user.getPassword(),
                user.getRoles()
        );
    }
}

File: user/src/main/kotlin/com/kjj/user/dto/user/UserDto.kt
package com.kjj.user.dto.user

import com.kjj.user.entity.User

data class UserDto(
    val id: Long?,
    val username: String,
    val password: String,
    val roles: String
) {
    companion object {
        fun from(user: User): UserDto {
            return UserDto(
                user.id,
                user.username,
                user.password,
                user.roles
            )
        }
    }
}